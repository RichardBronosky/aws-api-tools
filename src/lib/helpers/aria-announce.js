const formatMsg = msg => {
  return '<div aria-live="assertive" aria-atomic="true">' + msg + '</div>'
}

export const taskDone = (fn, time, id) => {
  let timers = {}
  if (typeof fn == 'function') {
    if (timers[id]) clearTimeout(timers[id])
    timers[id] = setTimeout(fn, time)
  }
}

export const announce = (msg, form, ele, delay) => {
  if (typeof ele == 'object') {
    msg += ele.innerText.replace(/\n/g, ' ')
  }
  if (document) {
    document.getElementById('aria-announce').innerHTML = ''
    taskDone(
      () => {
        document.getElementById('aria-announce').innerHTML = formatMsg(msg)
      },
      delay || 500,
      'announceMessage'
    )
    if (form) {
      taskDone(
        () => {
          form.querySelectorAll('[aria-invalid="true"]')[0].focus()
        },
        1000,
        'focusElement'
      )
    }
  }
}
