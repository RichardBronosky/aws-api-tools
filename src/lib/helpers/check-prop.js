export const checkProp = (prop, noProp, value) => {
  return prop ? (prop && noProp ? null : prop) : !noProp ? value || prop : null
}
