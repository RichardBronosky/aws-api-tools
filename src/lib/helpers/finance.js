import { getFromBrowserStorage } from './storage'
import { getCurrentLocation } from './geo-location'
import { getCartTotal, getCart } from './cart'
import { getRemainingTotal } from './checkout/global'

export const getFinancingPriority = (financePlans, total, noThreshold) => {
  const order = getFromBrowserStorage('session', 'order')
  if (order && order.promotions && order.promotions.totalSavings) {
    total -= order.promotions.totalSavings
  }
  let newFinancePlans = []
  let financePlanCodes = []
  for (let i = 0, n = financePlans.length; i < n; i++) {
    if (!financePlanCodes.includes(financePlans[i].financeCode)) {
      financePlanCodes.push(financePlans[i].financeCode)
    }
  }
  for (let i = 0, n = financePlanCodes.length; i < n; i++) {
    let priority = 0
    let index = 0
    const plans = financePlans.filter(plan => plan.financeCode === financePlanCodes[i])
    for (let x = 0, y = plans.length; x < y; x++) {
      const planPriority = parseInt(plans[x].priority)
      if (priority < planPriority) {
        priority = planPriority
        index = x
      }
    }
    if (
      ((total ? total : (order && order.total) || 0) >= plans[index].threshold || noThreshold) &&
      newFinancePlans.filter(plan => plan.financeCode === plans[index].financeCode).length < 1
    ) {
      if (plans[index].downPaymentRequired) {
        newFinancePlans.unshift(plans[index])
      } else if (!plans[index].downPaymentRequired) {
        newFinancePlans.push(plans[index])
      }
    }
  }
  let downPaymentPlans = newFinancePlans.filter(plan => plan.downPaymentRequired)
  let normalPlans = newFinancePlans.filter(plan => !plan.downPaymentRequired)
  let primaryDownPayment, primaryNormal
  if (downPaymentPlans.length > 1) {
    for (let i = 0, n = downPaymentPlans.length; i < n; i++) {
      primaryDownPayment = !primaryDownPayment
        ? downPaymentPlans[i]
        : downPaymentPlans[i].threshold > primaryDownPayment.threshold
        ? downPaymentPlans[i].threshold
        : primaryDownPayment
    }
  } else {
    primaryDownPayment = downPaymentPlans[0]
  }
  if (normalPlans.length > 1) {
    for (let i = 0, n = normalPlans.length; i < n; i++) {
      primaryNormal = !primaryNormal
        ? normalPlans[i]
        : normalPlans[i].threshold > primaryNormal.threshold
        ? normalPlans[i].threshold
        : primaryNormal
    }
  } else {
    primaryNormal = normalPlans[0]
  }
  return [primaryDownPayment, primaryNormal]
}

export const getFinancePlansFromRegion = () => {
  const financePlanFeed = require('../generators/data/finance-plan-feed.json')
  let financePlans = []
  if (financePlanFeed) {
    const loc = getCurrentLocation()
    if (loc) {
      financePlans = financePlanFeed[`${ loc.region.toLowerCase() }${ loc.price_zone }`]
        ? financePlanFeed[`${ loc.region.toLowerCase() }${ loc.price_zone }`]
        : financePlanFeed[`${ loc.region.toLowerCase() }0`] || financePlanFeed['fl0']
    } else {
      financePlans = financePlanFeed['fl0']
    }
  }
  return financePlans
}

export const getFinancePlans = (price = 0, cart = false, noThreshold = false) => {
  let realFinancePlans = []
  const financePlans = getFinancePlansFromRegion()
  if (financePlans && financePlans.length > 0) {
    realFinancePlans = financePlans
    if (realFinancePlans && realFinancePlans.length > 0) {
      const cartTotal = getCartTotal(getCart())
      const orderValues = getRemainingTotal(false, cartTotal, false, false)
      const totalForFin = orderValues.total + price
      realFinancePlans = getFinancingPriority(financePlans, cart ? orderValues.total : totalForFin, noThreshold)
    }
  }
  return realFinancePlans
}

export const productFinancing = price => {
  let finance = { financeAmount: 0, showFinance: false }
  const realFinancePlans = getFinancePlans(price)
  if (realFinancePlans && realFinancePlans.length > 0) {
    const finForCalc = realFinancePlans.filter(plan => plan && typeof plan !== 'undefined' && plan.downPaymentRequired)
    if (finForCalc && finForCalc.length > 0) {
      let financeAmount = 0
      if (finForCalc[0].numberOfMonths > 0) {
        financeAmount = Math.ceil(price / parseInt(finForCalc[0].numberOfMonths))
      }
      finance = {
        financeAmount: financeAmount,
        showFinance: financeAmount >= 10,
      }
    }
  }
  return finance
}

export const getFinanceMarketingMessageData = () => {
  const financePlans = getFinancePlans()
  if (financePlans && financePlans.length > 0) {
    const downPayment = financePlans.filter(plan => plan && typeof plan !== 'undefined' && plan.downPaymentRequired)
    if (downPayment && downPayment.length > 0) {
      return {
        siteFriendlyLabel: downPayment[0].siteFriendlyLabel,
        marketingMessage:
          downPayment[0].marketingMessage &&
          downPayment[0].marketingMessage.markdown &&
          downPayment[0].marketingMessage.markdown.childMarkdownRemark &&
          downPayment[0].marketingMessage.markdown.childMarkdownRemark.html,
      }
    } else {
      return (
        financePlans[0] && {
          siteFriendlyLabel: financePlans[0].siteFriendlyLabel,
          marketingMessage:
            financePlans[0].marketingMessage &&
            financePlans[0].marketingMessage.markdown &&
            financePlans[0].marketingMessage.markdown.childMarkdownRemark &&
            financePlans[0].marketingMessage.markdown.childMarkdownRemark.html,
        }
      )
    }
  }
}
