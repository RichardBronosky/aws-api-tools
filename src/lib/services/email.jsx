const axios = require('axios')

export const emailSignup = body =>
  axios.post(`${ process.env.GATSBY_EMAIL_SERVICE_URL }/signup/`, body).then(response => response.data)
