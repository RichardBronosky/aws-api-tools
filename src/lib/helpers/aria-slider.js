import { taskDone } from './aria-announce'

export const sliderfix = (slider, type) => {
  let ele = slider

  const add = arr => {
    for (let ii = 0; ii < arr.length; ii++) {
      let anchor = arr[ii]
      anchor.setAttribute('tabindex', '-1')
      anchor.setAttribute('aria-hidden', 'true')
    }
  }

  const remove = arr => {
    for (let ii = 0; ii < arr.length; ii++) {
      let anchor = arr[ii]
      anchor.removeAttribute('tabindex')
      anchor.removeAttribute('aria-hidden')
    }
  }

  const check = () => {
    if (ele) {
      let parent = ele.current
      if (parent) {
        let slides = parent.querySelectorAll('.slick-slide')
        for (let i = 0; i < slides.length; i++) {
          let slide = slides[i]

          if (slide.classList.contains('slick-active')) {
            remove(slide.querySelectorAll('a'))
            remove(slide.querySelectorAll('button'))
          } else {
            add(slide.querySelectorAll('a'))
            add(slide.querySelectorAll('button'))
          }
        }
      }
    }
  }

  if (ele && ele.current) {
    taskDone(
      () => {
        check()
      },
      type == 'init' ? 1 : 200,
      'adjustSlides'
    )
  }
}
