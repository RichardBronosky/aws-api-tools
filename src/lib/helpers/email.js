import { emailSignup } from '../services/email'
import { getFromBrowserStorage } from './storage'
import { validateEmail, validateZip } from './string-helper'
import { getCurrentLocation } from './geo-location'
import { announce } from './aria-announce'
import { getLocation } from '../services/location'

export const emailSubscribe = (e, email, zip, error, setState) => {
  e.preventDefault()
  setState({
    loading: true,
    submitted: false,
  })
  const validEmail = validateEmail(email)
  const validZip = validateZip(zip)
  if (validEmail && validZip) {
    getLocation(zip)
      .then(loc => {
        const uuid = getFromBrowserStorage('local', 'uuid')
        const date = new Date().toISOString()
        const location = getCurrentLocation()
        emailSignup({
          email: email,
          sourcecode: 'ONLINE',
          sourcekey: uuid,
          subsourcecode: 'PROFILE',
          sourcedate: date,
          userid: uuid,
          zipcode: zip,
          storenumber:
            loc && loc.response && loc.response.distribution_index
              ? loc.response.distribution_index
              : location.distribution_index,
        })
          .then(() => {
            const message = 'Signed up for emails!'
            announce('Form Submitted: ' + message)
            setState({
              success: message,
              submitted: true,
              loading: false,
              email: '',
              zip: '',
            })
          })
          .catch(() => {
            const message = '*Unable to sign up for emails.'
            if (e.currentTarget && e.currentTarget.parentElement) {
              announce('Form Error: ' + message, e.currentTarget.parentElement)
            }
            setState({
              error: message,
              submitted: false,
              loading: false,
            })
          })
      })
      .catch(() => {
        const message = '*Unable to sign up for emails.'
        if (e.currentTarget && e.currentTarget.parentElement) {
          announce('Form Error: ' + message, e.currentTarget.parentElement)
        }
        setState({
          error: message,
          submitted: false,
          loading: false,
        })
      })
  } else {
    const check = () => {
      return !validEmail && !validZip
        ? '*Invalid email address and zip code'
        : !validEmail
        ? '*Invalid email address.'
        : '*Invalid zip code.'
    }
    if (e.currentTarget && e.currentTarget.parentElement) {
      announce('Form Error: ' + error, e.currentTarget.parentElement)
    }
    setState({
      error: check(),
      fields: [!validEmail, !validZip],
      submitted: false,
      loading: false,
    })
  }
}
