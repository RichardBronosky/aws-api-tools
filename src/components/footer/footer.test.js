import React from 'react'
import renderer from 'react-test-renderer'
import Footer from './footer'
import * as financeHelper from '@helpers/finance'
import * as emailHelper from '../../lib/helpers/email'

jest.mock('../shared/link', () => 'RTGLink')

beforeEach(() => {
  financeHelper.getFinancePlans = jest.fn(() => [{ promoMessage: "Test Promo Message" }])
})

describe('Footer', () => {
  const testProps = {
    activeMenu: '',
    email: '',
    zip: '',
    error: null,
    success: null,
    submitted: false,
    loading: false,
    fields: [false, false],
    setState: jest.fn(),
    node: {
      id: 'testid',
      linkLists: [
        {
          id: 'testlinklikstid',
          displayText: 'Test Link List',
          headingLink: {
            text: 'Test heading text',
          },
          links: [
            {
              id: 'testlinkid',
              displayText: 'Test Link',
            },
          ],
        },
      ],
      contact: {
        childMarkdownRemark: {
          html: <div>hi</div>,
        },
      },
      termsOfUse: null,
      privacyPolicy: null,
    },
  }

  it('renders correctly with default props', () => {
    const tree = renderer.create(<Footer { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls setState on email input change', () => {
    const tree = renderer.create(<Footer { ...testProps } />)
    tree.root.findByProps({ className: 'signup-email' }).props.onChange({ target: { value: 'hi' } })
    expect(testProps.setState).toHaveBeenCalled()
  })

  it('calls setState on zip input change', () => {
    const tree = renderer.create(<Footer { ...testProps } />)
    tree.root.findByProps({ className: 'signup-zip' }).props.onChange({ target: { value: 'hi' } })
    expect(testProps.setState).toHaveBeenCalled()
  })

  it('calls emailSubscribe on submit click', () => {
    const tree = renderer.create(<Footer { ...testProps } />)
    emailHelper.emailSubscribe = jest.fn()
    tree.root.findByProps({ className: 'signup-btn' }).props.onClick()
    expect(emailHelper.emailSubscribe).toHaveBeenCalled()
  })
})
