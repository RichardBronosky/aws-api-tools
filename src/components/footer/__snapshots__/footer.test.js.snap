// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly with default props 1`] = `
<footer>
  <div
    className="grid-container"
  >
    <div
      className="bottom-row "
    >
      <div>
        <div
          className="left"
        >
          <div
            className="grid-x"
          >
            <div
              className="small-12 medium-12 large-9 grid-x"
            >
              <p
                className="cc-text"
              >
                <img
                  alt="Rooms To Go Credit"
                  className="cc-img"
                  src="undefinedundefinedrtg-cc.png?cache-id=undefined"
                />
                With a Rooms To Go Credit Card, you can take advantage of exclusive financing offers in your area.
                 
              </p>
            </div>
            <div
              className="cell small-12 medium-12 large-3 grid-x"
            >
              <RTGLink
                aria-label="Learn More about Rooms To Go Credit Cards"
                className="credit-learn-more blue-action-btn small-12"
                data={
                  Object {
                    "action": "apply-now",
                    "category": "credit-card",
                    "displayText": "Apply Now",
                    "id": "apply-now",
                    "label": "footer",
                    "title": "Rooms To Go Credit Options",
                    "url": undefined,
                  }
                }
                target="_blank"
              >
                Apply Now
              </RTGLink>
              <button
                aria-label="Learn More about Rooms To Go Credit Cards"
                className="info-modal credit-learn-more link"
                gtm-action="learn more"
                gtm-category="credit-card"
                gtm-label="footer"
                onClick={[Function]}
              >
                Learn More
              </button>
            </div>
          </div>
        </div>
        <div
          className="right"
        >
          <div
            className="email-label"
          >
            Enter email and zip code to receive special offers and more
          </div>
          <form>
            <div
              className="email-container"
            >
              <label
                className="signup-email-label"
                htmlFor="signup_email"
              >
                Email Address:
              </label>
              <input
                aria-invalid={false}
                aria-required="true"
                className="signup-email"
                id="signup_email"
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="zip-container"
            >
              <label
                className="signup-zip-label"
                htmlFor="zip"
              >
                Zip Code:
              </label>
              <input
                aria-invalid={false}
                aria-required="true"
                className="signup-zip"
                id="zip"
                name="zip"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <button
              className="signup-btn"
              gtm-action="email signup"
              gtm-category="footer"
              id="signup-btn"
              name="signup_btn"
              onClick={[Function]}
              value="Sign Up"
            >
              Sign Up
            </button>
          </form>
        </div>
      </div>
    </div>
    <div>
      <nav
        aria-label="Footer Links"
      >
        <div
          className="menu-col"
        >
          <div
            className="heading"
          >
            Test heading text
            <i
              className="menu-inactive"
            />
          </div>
          <div
            className=" sub-menu"
          >
            <RTGLink
              action="link click"
              category="footer"
              data={
                Object {
                  "displayText": "Test Link",
                  "id": "testlinkid",
                }
              }
              label="Test Link"
              noAriaLabel="true"
            />
          </div>
        </div>
        <div
          className="menu-col"
          dangerouslySetInnerHTML={
            Object {
              "__html": <div>
                hi
              </div>,
            }
          }
        />
      </nav>
    </div>
    <div
      className="social-media grid-container"
    >
      <div
        className="grid-x"
      >
        <div
          className="social-icons small-12  large-4"
        >
          <RTGLink
            className="icon"
            data={
              Object {
                "action": "social click",
                "category": "footer",
                "id": "facebook-icon",
                "label": "facebook",
                "url": "https://www.facebook.com/roomstogo/",
              }
            }
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Facebook (opens in new window)"
              className="fb"
              src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2023.94%2051.86%22%20fill%3D%22%230854A0%22%3E%3Ctitle%3Eicon-social-fb%3C%2Ftitle%3E%3Cg%20id%3D%22Layer%5f2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Objects%22%3E%3Cpath%20d%3D%22M5.43%2C51.86H15.87V25.71h7.29L23.94%2C17H15.87V12c0-2.07.42-2.88%2C2.41-2.88h5.66V0H16.71C8.93%2C0%2C5.43%2C3.42%2C5.43%2C10v7H0v8.87H5.43Z%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
            />
          </RTGLink>
          <RTGLink
            className="icon"
            data={
              Object {
                "action": "social click",
                "category": "footer",
                "id": "twitter-icon",
                "label": "twitter",
                "url": "https://twitter.com/RoomsToGo/",
              }
            }
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Twitter (opens in new window)"
              src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2045.71%2037.15%22%20fill%3D%22%230854A0%22%3E%3Ctitle%3Eicon-social-tw%3C%2Ftitle%3E%3Cg%20id%3D%22Layer%5f2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Objects%22%3E%3Cpath%20d%3D%22M0%2C32.93a26.56%2C26.56%2C0%2C0%2C0%2C14.38%2C4.22c17.41%2C0%2C27.24-14.71%2C26.65-27.9A19.1%2C19.1%2C0%2C0%2C0%2C45.71%2C4.4a18.7%2C18.7%2C0%2C0%2C1-5.39%2C1.47A9.37%2C9.37%2C0%2C0%2C0%2C44.45.68a18.71%2C18.71%2C0%2C0%2C1-6%2C2.28%2C9.38%2C9.38%2C0%2C0%2C0-16%2C8.55A26.64%2C26.64%2C0%2C0%2C1%2C3.18%2C1.72%2C9.39%2C9.39%2C0%2C0%2C0%2C6.09%2C14.24a9.25%2C9.25%2C0%2C0%2C1-4.25-1.18%2C9.39%2C9.39%2C0%2C0%2C0%2C7.52%2C9.32%2C9.42%2C9.42%2C0%2C0%2C1-4.24.16%2C9.39%2C9.39%2C0%2C0%2C0%2C8.76%2C6.51A18.82%2C18.82%2C0%2C0%2C1%2C0%2C32.93Z%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
            />
          </RTGLink>
          <RTGLink
            className="icon"
            data={
              Object {
                "action": "social click",
                "category": "footer",
                "id": "instagram-icon",
                "label": "instagram",
                "url": "https://www.instagram.com/roomstogo/",
              }
            }
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Instagram (opens in new window)"
              src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2047.04%2047.04%22%20fill%3D%22%230854A0%22%3E%3Ctitle%3Eicon-social-ins%3C%2Ftitle%3E%3Cg%20id%3D%22Layer%5f2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Objects%22%3E%3Cpath%20d%3D%22M23.52%2C4.24c6.28%2C0%2C7%2C0%2C9.5.13a13.2%2C13.2%2C0%2C0%2C1%2C4.37.81A7.43%2C7.43%2C0%2C0%2C1%2C40.1%2C6.94a7.28%2C7.28%2C0%2C0%2C1%2C1.75%2C2.71A12.68%2C12.68%2C0%2C0%2C1%2C42.66%2C14c.12%2C2.48.14%2C3.22.14%2C9.5s0%2C7-.14%2C9.5a12.68%2C12.68%2C0%2C0%2C1-.81%2C4.37%2C7.67%2C7.67%2C0%2C0%2C1-4.46%2C4.46%2C12.68%2C12.68%2C0%2C0%2C1-4.37.81c-2.48.12-3.22.14-9.5.14s-7%2C0-9.5-.14a12.68%2C12.68%2C0%2C0%2C1-4.37-.81A7.28%2C7.28%2C0%2C0%2C1%2C6.94%2C40.1a7.43%2C7.43%2C0%2C0%2C1-1.76-2.71A13.2%2C13.2%2C0%2C0%2C1%2C4.37%2C33c-.11-2.48-.13-3.22-.13-9.5s0-7%2C.13-9.5a13.2%2C13.2%2C0%2C0%2C1%2C.81-4.37A7.77%2C7.77%2C0%2C0%2C1%2C9.65%2C5.18%2C13.2%2C13.2%2C0%2C0%2C1%2C14%2C4.37c2.48-.11%2C3.22-.13%2C9.5-.13m0-4.24c-6.39%2C0-7.19%2C0-9.7.14A17.51%2C17.51%2C0%2C0%2C0%2C8.11%2C1.23%2C11.69%2C11.69%2C0%2C0%2C0%2C4%2C4%2C11.69%2C11.69%2C0%2C0%2C0%2C1.23%2C8.11%2C17.51%2C17.51%2C0%2C0%2C0%2C.14%2C13.82C0%2C16.33%2C0%2C17.13%2C0%2C23.52s0%2C7.19.14%2C9.7a17.51%2C17.51%2C0%2C0%2C0%2C1.09%2C5.71A11.69%2C11.69%2C0%2C0%2C0%2C4%2C43.09%2C11.53%2C11.53%2C0%2C0%2C0%2C8.11%2C45.8a17.22%2C17.22%2C0%2C0%2C0%2C5.71%2C1.1c2.51.11%2C3.31.14%2C9.7.14s7.19%2C0%2C9.7-.14a17.22%2C17.22%2C0%2C0%2C0%2C5.71-1.1%2C12%2C12%2C0%2C0%2C0%2C6.87-6.87%2C17.22%2C17.22%2C0%2C0%2C0%2C1.1-5.71c.11-2.51.14-3.31.14-9.7s0-7.19-.14-9.7a17.22%2C17.22%2C0%2C0%2C0-1.1-5.71A11.53%2C11.53%2C0%2C0%2C0%2C43.09%2C4a11.69%2C11.69%2C0%2C0%2C0-4.16-2.72A17.51%2C17.51%2C0%2C0%2C0%2C33.22.14C30.71%2C0%2C29.91%2C0%2C23.52%2C0%22%2F%3E%3Cpath%20d%3D%22M23.52%2C11.44A12.08%2C12.08%2C0%2C1%2C0%2C35.6%2C23.52%2C12.08%2C12.08%2C0%2C0%2C0%2C23.52%2C11.44m0%2C19.92a7.84%2C7.84%2C0%2C1%2C1%2C7.84-7.84%2C7.84%2C7.84%2C0%2C0%2C1-7.84%2C7.84%22%2F%3E%3Cpath%20d%3D%22M38.9%2C11a2.83%2C2.83%2C0%2C1%2C1-2.83-2.82A2.83%2C2.83%2C0%2C0%2C1%2C38.9%2C11%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
            />
          </RTGLink>
          <RTGLink
            className="icon pin"
            data={
              Object {
                "action": "social click",
                "category": "footer",
                "id": "pintrest-icon",
                "label": "pintrest",
                "url": "https://www.pinterest.com/roomstogo/",
              }
            }
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Pintrest (opens in new window)"
              src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2039.44%2050.95%22%20fill%3D%22%230854A0%22%3E%3Ctitle%3Eicon-social-pin%3C%2Ftitle%3E%3Cg%20id%3D%22Layer%5f2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Objects%22%3E%3Cpath%20d%3D%22M6%2C29.46a1%2C1%2C0%2C0%2C0%2C1.47-.73c.13-.51.45-1.81.6-2.35a1.42%2C1.42%2C0%2C0%2C0-.43-1.63A8.48%2C8.48%2C0%2C0%2C1%2C5.7%2C19c0-7.4%2C5.54-14%2C14.41-14C28%2C5%2C32.29%2C9.79%2C32.29%2C16.21c0%2C8.44-3.74%2C15.56-9.28%2C15.56a4.53%2C4.53%2C0%2C0%2C1-4.62-5.64C19.27%2C22.42%2C21%2C18.42%2C21%2C15.75c0-2.4-1.28-4.4-3.94-4.4-3.13%2C0-5.65%2C3.24-5.65%2C7.58a11.2%2C11.2%2C0%2C0%2C0%2C.94%2C4.63L8.55%2C39.5C7.43%2C44.24%2C8.38%2C50%2C8.47%2C50.62a.39.39%2C0%2C0%2C0%2C.69.17c.29-.37%2C4-5%2C5.28-9.57.36-1.31%2C2.06-8%2C2.06-8%2C1%2C1.94%2C4%2C3.65%2C7.15%2C3.65%2C9.41%2C0%2C15.79-8.58%2C15.79-20.06C39.44%2C8.08%2C32.09%2C0%2C20.91%2C0%2C7%2C0%2C0%2C10%2C0%2C18.28%2C0%2C23.31%2C1.91%2C27.79%2C6%2C29.46Z%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
            />
          </RTGLink>
          <RTGLink
            className="icon"
            data={
              Object {
                "action": "social click",
                "category": "footer",
                "id": "youtube-icon",
                "label": "youtube",
                "url": "https://www.youtube.com/channel/UCd77CqGAnGsyDD-lFFkyrNw",
              }
            }
            rel="noopener"
            target="_blank"
          >
            <img
              alt="YouTube (opens in new window)"
              src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2044.38%2051.22%22%20fill%3D%22%230854A0%22%3E%3Ctitle%3Eicon-social-youtube%3C%2Ftitle%3E%3Cg%20id%3D%22Layer%5f2%22%20data-name%3D%22Layer%202%22%3E%3Cg%20id%3D%22Objects%22%3E%3Cpath%20d%3D%22M28.67%2C20a4.3%2C4.3%2C0%2C0%2C0%2C3.14-1.83v1.62h2.71V5.19H31.81V16.25c-.33.42-1.07%2C1.09-1.6%2C1.09s-.72-.39-.72-1V5.18H26.78V17.37C26.78%2C18.81%2C27.22%2C20%2C28.67%2C20Z%22%2F%3E%3Cpath%20d%3D%22M16.61%2C16.13c0%2C2.57%2C1.34%2C3.9%2C4%2C3.9a3.71%2C3.71%2C0%2C0%2C0%2C3.9-3.9V9a3.8%2C3.8%2C0%2C0%2C0-3.9-3.91%2C3.72%2C3.72%2C0%2C0%2C0-4%2C3.91ZM19.4%2C9.22c0-.8.37-1.39%2C1.13-1.39s1.19.58%2C1.19%2C1.39V16c0%2C.79-.41%2C1.37-1.14%2C1.37A1.22%2C1.22%2C0%2C0%2C1%2C19.4%2C16Z%22%2F%3E%3Cpolygon%20points%3D%2210.14%2019.77%2013.2%2019.77%2013.2%2011.75%2016.75%200%2013.65%200%2011.69%207.9%209.58%200%206.51%200%2010.14%2011.75%2010.14%2019.77%22%2F%3E%3Cpath%20d%3D%22M38.17%2C23.08h-32A6.22%2C6.22%2C0%2C0%2C0%2C0%2C29.3V45a6.22%2C6.22%2C0%2C0%2C0%2C6.21%2C6.22h32A6.22%2C6.22%2C0%2C0%2C0%2C44.38%2C45V29.3A6.22%2C6.22%2C0%2C0%2C0%2C38.17%2C23.08ZM10.47%2C46.56H7.57V30.47h-3V27.74h8.92v2.73h-3Zm10.34%2C0H18.23V45a5.44%2C5.44%2C0%2C0%2C1-1.48%2C1.29c-1.39.8-3.3.78-3.3-2V32.68H16V43.32c0%2C.56.13.94.68.94s1.21-.65%2C1.52-1V32.68h2.58Zm9.94-2.87c0%2C1.71-.64%2C3.05-2.36%2C3.05a3%2C3%2C0%2C0%2C1-2.45-1.24v1.06H23.33V27.74h2.61V33.8a3.09%2C3.09%2C0%2C0%2C1%2C2.29-1.3c1.89%2C0%2C2.52%2C1.6%2C2.52%2C3.48ZM40.29%2C40H35.36v2.62c0%2C1%2C.08%2C1.94%2C1.12%2C1.94s1.16-.74%2C1.16-1.94v-1h2.65v1c0%2C2.67-1.15%2C4.29-3.87%2C4.29-2.46%2C0-3.72-1.8-3.72-4.29V36.41a3.81%2C3.81%2C0%2C0%2C1%2C3.92-4.08c2.48%2C0%2C3.67%2C1.57%2C3.67%2C4.08Z%22%2F%3E%3Cpath%20d%3D%22M36.51%2C34.76c-1%2C0-1.15.67-1.15%2C1.63v1.4h2.28v-1.4C37.64%2C35.45%2C37.44%2C34.76%2C36.51%2C34.76Z%22%2F%3E%3Cpath%20d%3D%22M26.47%2C34.84a2%2C2%2C0%2C0%2C0-.53.43v8.65a2.15%2C2.15%2C0%2C0%2C0%2C.61.49%2C1.1%2C1.1%2C0%2C0%2C0%2C1.32-.14%2C1.62%2C1.62%2C0%2C0%2C0%2C.22-1V36.14a1.78%2C1.78%2C0%2C0%2C0-.27-1.07A1.09%2C1.09%2C0%2C0%2C0%2C26.47%2C34.84Z%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
            />
          </RTGLink>
        </div>
        <div
          className="get-app small-12  large-4"
        >
          <RTGLink
            className="icon"
            data={
              Object {
                "action": "app download",
                "category": "footer",
                "displayText": "Get the Rooms To Go Coupon App",
                "id": "coupon-app",
                "label": "app download",
                "url": "https://www.roomstogocoupons.com/",
              }
            }
          >
            <img
              alt="Get the Rooms To Go Coupon App"
              src="data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 46 86' fill='rgb%28255,255,255%29'%3E%3Ctitle%3Eicon-app%3C/title%3E%3Cg id='Layer_2' data-name='Layer 2'%3E%3Cg id='Layer_1-2' data-name='Layer 1'%3E%3Cpath d='M6,0A6,6,0,0,0,0,6V80a6,6,0,0,0,6,6H40a6,6,0,0,0,6-6V6a6,6,0,0,0-6-6ZM4,10H42V70H4ZM23,73a5,5,0,1,1-5,5A5,5,0,0,1,23,73Z'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E"
            />
          </RTGLink>
          <span
            aria-hidden="true"
            style={
              Object {
                "display": "block",
              }
            }
          >
            Get the Rooms To Go Coupon App
          </span>
        </div>
        <div
          className="payment-methods small-12 large-4"
        >
          <img
            alt="Accepted payment types: Discover, Mastercard, Visa, Amex, Rooms to go Credit, PayPal, Affirm"
            className="paymentLogos"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="blue-row"
  >
    <div
      className="grid-container grid-margin-y"
    >
      <div
        className="rtg-logo grid-x"
      >
        <RTGLink
          className="icon pin small-3"
          data={
            Object {
              "action": "click",
              "category": "footer",
              "label": "logo",
              "slug": "/",
              "title": "Rooms To Go",
            }
          }
        >
          <img
            alt="Rooms To Go"
            className="desktop"
            src="undefinedundefinedmobile-logo_transparent.png?cache-id=undefined&h=250"
          />
          <img
            alt="Rooms To Go"
            className="mobile"
            src="undefinedundefinedmobile-logo_transparent.png?cache-id=undefined&h=250"
          />
        </RTGLink>
      </div>
      <div
        className="notices"
      >
        <p
          className="first"
        >
          <RTGLink
            action="privacy click"
            category="footer"
            data={null}
            label="privacy policy"
            noAriaLabel="true"
          />
            |  
          <RTGLink
            action="terms click"
            category="footer"
            data={null}
            label="terms of use"
            noAriaLabel="true"
          />
        </p>
        <div
          className="copyright"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
      </div>
      <div
        className="logos"
      >
        <img
          alt="Norton Secured Powered by digicert"
          src="undefinedundefinedlogo-norton_transparent.png?cache-id=undefined&h=250"
        />
        <img
          alt="BBB Accredited Business A+"
          src="undefinedundefinedlogo-bbb_transparent.png?cache-id=undefined&h=250"
        />
      </div>
    </div>
  </div>
</footer>
`;
